### PR TITLE
feat: Adding auth service

### PR DESCRIPTION
**Pull request by JigarJoshi**
To proxy / implement oauth provider.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,7 @@ ${API_DIR}/client/${V}/%/http.go: ${PROTO_DIR}/%_openapi.yaml
 		-o ${API_DIR}/client/${V}/$(*F)/http.go \
 		${PROTO_DIR}/$(*F)_openapi.yaml
 
-generate: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${GEN_DIR}/health.pb.go ${GEN_DIR}/health.pb.gw.go ${GEN_DIR}/admin.pb.go ${GEN_DIR}/admin.pb.gw.go ${PROTO_DIR}/api_openapi.yaml ${PROTO_DIR}/admin_openapi.yaml
+generate: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${GEN_DIR}/health.pb.go ${GEN_DIR}/health.pb.gw.go ${GEN_DIR}/admin.pb.go ${GEN_DIR}/admin.pb.gw.go ${GEN_DIR}/auth.pb.go ${GEN_DIR}/auth.pb.gw.go ${PROTO_DIR}/api_openapi.yaml ${PROTO_DIR}/admin_openapi.yaml ${PROTO_DIR}/auth_openapi.yaml
```
Comment (efirs) [inline Makefile:31]: This line is getting to long.
Let's break it be service.

Comment (JigarJoshi) [reply Makefile]: sounds good. fixed.

File: Makefile [hunk 2]
```diff
@@ -28,7 +28,13 @@ ${API_DIR}/client/${V}/%/http.go: ${PROTO_DIR}/%_openapi.yaml
 		-o ${API_DIR}/client/${V}/$(*F)/http.go \
 		${PROTO_DIR}/$(*F)_openapi.yaml
 
-generate: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${GEN_DIR}/health.pb.go ${GEN_DIR}/health.pb.gw.go ${GEN_DIR}/admin.pb.go ${GEN_DIR}/admin.pb.gw.go ${PROTO_DIR}/api_openapi.yaml ${PROTO_DIR}/admin_openapi.yaml
+generate: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go \
+${GEN_DIR}/health.pb.go ${GEN_DIR}/health.pb.gw.go \
+${GEN_DIR}/admin.pb.go ${GEN_DIR}/admin.pb.gw.go \
+${GEN_DIR}/auth.pb.go ${GEN_DIR}/auth.pb.gw.go \
+${PROTO_DIR}/api_openapi.yaml \
```
Comment (efirs) [inline Makefile:35]: Can this three go to it corresponding service above?
Like this:
```
${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${PROTO_DIR}/api_openapi.yaml
```

Comment (JigarJoshi) [reply Makefile]: updated the formatting.

File: Makefile [hunk 3]
```diff
@@ -28,7 +28,10 @@ ${API_DIR}/client/${V}/%/http.go: ${PROTO_DIR}/%_openapi.yaml
 		-o ${API_DIR}/client/${V}/$(*F)/http.go \
 		${PROTO_DIR}/$(*F)_openapi.yaml
 
-generate: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${GEN_DIR}/health.pb.go ${GEN_DIR}/health.pb.gw.go ${GEN_DIR}/admin.pb.go ${GEN_DIR}/admin.pb.gw.go ${PROTO_DIR}/api_openapi.yaml ${PROTO_DIR}/admin_openapi.yaml
+generate: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${PROTO_DIR}/api_openapi.yaml \
```
Comment (rbarabas) [inline Makefile:31]: How about refactoring this target a bit like this:

```
$ cat Makefile
COMPONENTS = admin auth api health

generate:
	for COMPONENT in $(COMPONENTS); do \
		echo $${COMPONENT}.pb.go $${COMPONENT}.pb.gw.go $${COMPONENT}.yaml; \
	done
	
```

```
$ make generate
for COMPONENT in admin auth api health; do \
		echo ${COMPONENT}.pb.go ${COMPONENT}.pb.gw.go ${COMPONENT}.yaml; \
	done
admin.pb.go admin.pb.gw.go admin.yaml
auth.pb.go auth.pb.gw.go auth.yaml
api.pb.go api.pb.gw.go api.yaml
health.pb.go health.pb.gw.go health.yaml
```

**Review by rbarabas — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (rbarabas) [reply Makefile:31]: @efirs FYI

**Review by efirs — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (efirs) [reply Makefile:31]: Good idea, let's remember this and I'll fix it once I touch this code block next time.

**Review by rbarabas — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (JigarJoshi) [reply Makefile:31]: Idea is great but implementation isn't just as simple as this. I will add this improvement later.

**Review by rbarabas — APPROVED** [review]

**Pull request merged**